config/shared_config.go

```diff
@@ -210,78 +259,478 @@ func loadSharedConfig(ctx context.Context, configs configs) (Config, error) {
+	configSections, err := loadIniFiles(options.ConfigurationFiles)
 	if err != nil {
 		return SharedConfig{}, err
 	}
 
+	// check for profile prefix and drop duplicates or invalid profiles
+	err = processConfigSections(ctx, configSections, options.Logger)
+	if err != nil {
+		return SharedConfig{}, err
+	}
+
+	// load shared credentials sections from shared credentials INI options
+	credentialsSections, err := loadIniFiles(options.CredentialsFiles)
+	if err != nil {
+		return SharedConfig{}, err
+	}
+
+	// check for profile prefix and drop duplicates or invalid profiles
+	err = processCredentialsSections(ctx, credentialsSections, options.Logger)
+	if err != nil {
+		return SharedConfig{}, err
+	}
+
+	err = mergeSections(configSections, credentialsSections)
+	if err != nil {
+		return SharedConfig{}, err
+	}
+
+	// profile should be lower-cased to standardize
+	profile = strings.ToLower(profile)
+
 	cfg := SharedConfig{}
 	profiles := map[string]struct{}{}
-	if err = cfg.setFromIniFiles(profiles, profile, files); err != nil {
+	if err = cfg.setFromIniSections(profiles, profile, configSections, options.Logger); err != nil {
 		return SharedConfig{}, err
 	}
 
 	return cfg, nil
 }
 
-type sharedConfigFile struct {
+func processConfigSections(ctx context.Context, sections ini.Sections, logger logging.Logger) error {
+
+	prefix := "profile "
+	for _, section := range sections.List() {
+		// drop profiles without prefix for config files
+		if !strings.HasPrefix(section, prefix) && !strings.EqualFold(section, "default") {
+			// drop this section, as invalid profile name
+			sections.DeleteSection(section)
+
+			if logger != nil {
+				logger.Logf(logging.Debug,
+					"profile %v is ignored. A non-default profile without the \"profile \" prefix is invalid "+
+						"for the shared configuration file.\n",
+					section,
+				)
+			}
+		}
+	}
+
+	// rename sections to remove `profile ` prefixing to match with credentials file.
+	// if default is already present, it will be dropped.
+	for _, section := range sections.List() {
+		if strings.HasPrefix(section, prefix) {
+			v, ok := sections.GetSection(section)
+			if !ok {
+				return fmt.Errorf("error processing profiles within the shared configuration files")
+			}
+
+			// delete section with profile as prefix
+			sections.DeleteSection(section)
+
+			// set the value to non-prefixed name in sections.
+			section = strings.TrimPrefix(section, prefix)
+			if sections.HasSection(section) {
+				oldSection, _ := sections.GetSection(section)
+				v.Logs = append(v.Logs,
+					fmt.Sprintf("A default profile prefixed with `profile` found in %s, "+
+						"overrided non-prefixed default profile from %s", v.SourceFile, oldSection.SourceFile))
+			}
+
+			// assign non-prefixed name to section
+			v.Name = section
+			sections.SetSection(section, v)
+		}
+	}
+	return nil
+}
+
+func processCredentialsSections(ctx context.Context, sections ini.Sections, logger logging.Logger) error {
```

Comment:
What happens if section is `[profile]` ?   Assuming this would be considered a valid name. Probably want special case test for credentials file with that name, to ensure its not broken later.